source/code/plugin/health/health_monitor_telemetry.rb

```diff
@@ -8,31 +8,50 @@
 module HealthModel
     class HealthMonitorTelemetry
 
-        attr_reader :monitor_records
+        attr_reader :monitor_records, :last_sent_time
+        @@TELEMETRY_SEND_INTERVAL = 60
+
+        def initialize
+            @last_sent_time = Time.now
+        end
 
         def send
-            log = HealthMonitorHelpers.get_log_handle
-            log.info "Sending #{@monitor_records.size} state change events"
-            @monitor_records.each{|record|
-                ApplicationInsightsUtility.sendCustomEvent("HealthMonitorStateChangeEvent", {
-                    "MonitorId" => record['monitor_id'],
-                    "OldState" => record['old_state'],
-                    "NewState" => record['new_state']
-                })
-            }
-            @monitor_records = []
+            if Time.now > @last_sent_time + @@TELEMETRY_SEND_INTERVAL * 60
+                log = HealthMonitorHelpers.get_log_handle
+                log.info "Sending #{@monitor_records.size} state change events"
+                if @monitor_records.size > 0
+                    hash_to_send = {}
+                    @monitor_records.each{|k,v|
+                        v.each{|k1,v1|
+                            hash_to_send["#{k}-#{k1}"] = v1
+                        }
+                    }
+                    ApplicationInsightsUtility.sendCustomEvent("HealthMonitorStateChangeEvent", hash_to_send)
+                end
+                @monitor_records = {}
```

Comment:
what will happen if add_monitor_to_telemetry & send happens at the same time (one adds & other clears)